language-server/src/features/document-settings.js

```diff
@@ -39,19 +35,21 @@ export default {
 };
 
 const documentSettings = new Map();
-let globalSettings = {};
+const defaultSettings = {
+  schemaFilePatterns: ["**/*.schema.json", "**/schema.json"]
+};
 
 export const getDocumentSettings = async (connection, uri) => {
   if (!hasConfigurationCapability) {
-    return globalSettings;
+    return defaultSettings;
   }
 
   if (!documentSettings.has(uri)) {
     const result = await connection.workspace.getConfiguration({
       scopeUri: uri,
       section: "jsonSchemaLanguageServer"
     });
-    documentSettings.set(uri, result ?? globalSettings);
+    documentSettings.set(uri, result ?? defaultSettings);
```

Comment:
I don't think this is going to be enough. If my settings are set to `{ "defaultDialect": "https://json-schema.org/draft/2020-12/schema" }`, then the `??` doesn't apply and the `defaultSettings` don't get set. I think we need something like,

```suggestion
    documentSettings.set(uri, { ...defaultSettings, ...(result ?? {}) });
```

That way it merges the settings that are retrieved with the `defualtSettings`.

Reply:
Ah yes, my bad. I was only thinking about `schemaFilePatterns`.

Reply:
Fixed this and the recordings are also uploaded for both vscode and neovim.